api/src/main/java/uk/gov/hmcts/payment/api/controllers/CasePaymentOrdersController.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoBadRequestException;
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoClientException;
+import uk.gov.hmcts.payment.casepaymentorders.client.exceptions.CpoInternalServerErrorException;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.INTERNAL_SERVER_ERROR;
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+/**
+ * Case payment orders controller
+ */
+
+@RestController
+@Api(tags = {"Case Payment Order"})
+@SwaggerDefinition(tags = {@Tag(name = "CasePaymentOrderController", description = "Case Payment Order REST API")})
+public class CasePaymentOrdersController {
+    private static final Logger LOG = LoggerFactory.getLogger(CasePaymentOrdersController.class);
+
+    private final CasePaymentOrdersService casePaymentOrdersService;
+    private final CasePaymentOrdersMapper casePaymentOrdersMapper;
+
+    @Autowired
+    public CasePaymentOrdersController(CasePaymentOrdersService casePaymentOrdersService,
+                                       CasePaymentOrdersMapper casePaymentOrdersMapper) {
+        this.casePaymentOrdersService = casePaymentOrdersService;
+        this.casePaymentOrdersMapper = casePaymentOrdersMapper;
+    }
+
+    @ApiOperation(value = "Get payment orders for a case", notes = "Get payment orders for a case")
+    @ApiResponses(value = {
+        @ApiResponse(code = 200, message = "Payment orders retrieved"),
+        @ApiResponse(code = 400, message = "Bad request"),
+        @ApiResponse(code = 401, message = "Credentials are required to access this resource"),
+        @ApiResponse(code = 403, message = "Case Payment Orders info forbidden"),
+        @ApiResponse(code = 500, message = "Downstream system error")
+    })
+    @RequestMapping(value = "/case-payment-orders", method = GET)
+    public CasePaymentOrdersDto retrieveCasePaymentOrders(
+        @ApiParam(value = "Coma separated list of case ids.")
+        @RequestParam(name = "case-ids", required = false) String caseIds,
+        @ApiParam(value = "Page number - 1-based indexed page number to be served.")
+        @RequestParam(name = "page-number", required = false) String pageNumber,
+        @ApiParam(value = "Page size - number of elements on the page.")
+        @RequestParam(name = "page-size", required = false) String pageSize,
+        @RequestHeader(HttpHeaders.AUTHORIZATION) String authorization) {
+
+        CpoGetResponse casePaymentOrders = casePaymentOrdersService
+            .getCasePaymentOrders(caseIds, pageNumber, pageSize, authorization);
+
+        return casePaymentOrdersMapper.toCasePaymentOrdersDto(casePaymentOrders);
+    }
+
+    @ExceptionHandler(value = {CpoBadRequestException.class})
+    public ResponseEntity<String> cpoBadRequestException(CpoBadRequestException e) {
+        LOG.error("Error while calling case payment orders", e);
```

Comment:
Can we log error specific to the exception

Reply:
CpoBadRequestException - Bad Request Error while fetching case payment orders
CpoInternalServerErrorException - Internal Server Error while fetching case payment orders
CpoClientException - Client Error while invoking case payment orders

The idea is to have identifiers/keywords to aggregate a specific error type from logs

Reply:
Updated error messages